package/gluon-mesh-vpn-wireguard/src/gluon-hex-to-b64.c

```diff
@@ -0,0 +1,102 @@
+// SPDX-FileCopyrightText: 2022 Jan-Niklas Burfeind <gluon@aiyionpri.me>
+// SPDX-License-Identifier: BSD-2-Clause
+// SPDX-FileContributor: read_hex() by Matthias Schiffer <mschiffer@universe-factory.net>
+
+#include <libubox/utils.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/** return how many samesized elements an array has */
+#define ARRAY_SIZE(arr) (sizeof(arr) / sizeof((arr)[0]))
+
+/**
+ * how many blocks should be encoded at once - can be configured
+ */
+#define BLOCK_AMOUNT 32
+
+/**
+ * smallest possible block size to encode in b64 without further contex
+ * is three bytes - do not change
+ */
+#define BLOCK_SIZE 3*BLOCK_AMOUNT
+
+/** print usage info and exit as failed */
+static void usage(void) {
+	fprintf(stderr, "Usage: gluon-hex-to-b64\n");
+	exit(1);
+}
+
+/**
+ * read a string of hexadecimal characters and return them as bytes
+ * return false in case any non-hexadecimal characters are provided
+ * return true on success
+ */
+static bool read_hex(uint8_t key[BLOCK_SIZE], const char *hexkey) {
+	if (strspn(hexkey, "0123456789abcdefABCDEF") != strlen(hexkey))
+		return false;
+
+	size_t i;
+	for (i = 0; i < BLOCK_SIZE; i++)
+		sscanf(&hexkey[2 * i], "%02hhx", &key[i]);
+
+	return true;
+}
+
+int main(int argc, char *argv[]) {
+	if (argc != 1)
+		usage();
+
+	unsigned char  hex_input[BLOCK_SIZE*2];
```

Comment:
```suggestion
	unsigned char hex_input[BLOCK_SIZE*2];
```

Reply:
resolved in 25753249e1dc3d5626511f7cc683cdfb575d0e20